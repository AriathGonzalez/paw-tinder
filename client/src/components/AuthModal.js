import { useState } from "react";
import { useNavigate } from "react-router-dom";
import { useCookies } from "react-cookie";
import axios from "axios";
import "../styles/AuthModal.css";

const AuthModal = ({ setShowModal, isSignUp, setIsSignUp }) => {
  const [email, setEmail] = useState(null);
  const [password, setPassword] = useState(null);
  const [confirmPassword, setConfirmPassword] = useState(null);
  const [error, setError] = useState(null);
  const [, setCookie] = useCookies(["user"]);

  let navigate = useNavigate();

  const handleClick = () => {
    setIsSignUp(true);
    setShowModal(false);
  };

  const handleSubmit = async (e) => {
    e.preventDefault();
    try {
      if (isSignUp && password !== confirmPassword) {
        setError("Passwords need to match!");
        return;
      }
      const response = await axios.post(
        `http://localhost:5000/users/${isSignUp ? "signup" : "login"}`,
        {
          email,
          password,
        }
      );

      setCookie("AuthToken", response.data.token);
      setCookie("UserId", response.data.userId);

      const success = response.status === 201;

      if (success && isSignUp) navigate("/onboarding");
      if (success && !isSignUp) navigate("/dashboard");

      window.location.reload();
    } catch (error) {
      console.log(error);
    }
  };

  return (
    <div className="auth-modal">
      <div className="close-icon" onClick={handleClick}>
        â“§
      </div>
      <h2>{isSignUp ? "Create Account" : "Log In"}</h2>
      <p>
        By Clicking Submit, you agree to our terms. Learn how we process your
        data in our Privacy Policy and Cookie Policy.
      </p>
      <form onSubmit={handleSubmit}>
        <input
          type="email"
          id="email"
          name="email"
          placeholder="email"
          required={true}
          onChange={(e) => setEmail(e.target.value)}
        />
        <input
          type="password"
          id="password"
          name="password"
          placeholder="password"
          required={true}
          onChange={(e) => setPassword(e.target.value)}
        />
        {isSignUp && (
          <input
            type="password"
            id="password-check"
            name="password-check"
            placeholder="confirm password"
            required={true}
            onChange={(e) => setConfirmPassword(e.target.value)}
          />
        )}
        <input className="secondary-button" type="submit" />
        <p>{error}</p>
      </form>
      <hr />
      <h2>GET THE APP</h2>
    </div>
  );
};

export default AuthModal;
