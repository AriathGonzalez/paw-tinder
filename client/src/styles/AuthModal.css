.auth-modal {
  background-color: rgba(249, 249, 249, 1);
  border-radius: 0.65em;
  display: flex;
  flex-direction: column;
  height: 36em;
  left: 50%;
  max-width: 20em;
  padding: 2.5em;
  position: fixed;
  top: 3em;
  transform: translateX(-50%);
  width: 20em;
}

.close-icon {
  align-self: end;
  cursor: pointer;
  font-size: 1.5rem;
}

.auth-modal form {
  display: flex;
  flex-direction: column;
}

.auth-modal input[type="password"],
.auth-modal input[type="email"] {
  margin: 0.5em;
  color: rgba(0, 184, 169, 0.75);
  font-size: 1rem;
  padding: 0.5em;
}
