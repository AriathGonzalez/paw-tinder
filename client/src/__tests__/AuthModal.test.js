import AuthModal from "../components/AuthModal";
import renderer from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";

describe("AuthModel Component", () => {
  describe("Snapshot Test", () => {
    it("Should render correctly", () => {
      const tree = renderer
        .create(
          <BrowserRouter>
            <AuthModal setShowModal={null} isSignUp={true} setIsSignUp={null} />
          </BrowserRouter>
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
