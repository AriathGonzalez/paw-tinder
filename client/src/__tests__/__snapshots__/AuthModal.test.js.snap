// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthModel Component Snapshot Test Should render correctly 1`] = `
<div
  className="auth-modal"
>
  <div
    className="close-icon"
    onClick={[Function]}
  >
    â“§
  </div>
  <h2>
    Create Account
  </h2>
  <p>
    By Clicking Submit, you agree to our terms. Learn how we process your data in our Privacy Policy and Cookie Policy.
  </p>
  <form
    onSubmit={[Function]}
  >
    <input
      id="email"
      name="email"
      onChange={[Function]}
      placeholder="email"
      required={true}
      type="email"
    />
    <input
      id="password"
      name="password"
      onChange={[Function]}
      placeholder="password"
      required={true}
      type="password"
    />
    <input
      id="password-check"
      name="password-check"
      onChange={[Function]}
      placeholder="confirm password"
      required={true}
      type="password"
    />
    <input
      className="secondary-button"
      type="submit"
    />
    <p />
  </form>
  <hr />
  <h2>
    GET THE APP
  </h2>
</div>
`;
